Adding two new operations for Reservations

kc786 opened the pull request:
1. Reservation Summaries
2. Reservation Details

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

kc786 committed:
Adding two new operations for Reservations
1. Reservation Summaries
2. Reservation Details

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -117,6 +117,127 @@
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationSummariesListResult"
```
would you want to add x-ms-pageable to this operation- https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-pageable 
You can set nextLinkName to null, which results in an enumerable for some of the SDKs.

kc786 replied:
pushed my changes

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -117,6 +117,127 @@
+          },  
+          {
+            "name": "grain",
+            "description": "Can be daily or monthly",
```
should it be an enum?

kc786 replied:
pushed my changes

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -117,6 +117,127 @@
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ReservationDetailsListResult"
```
same comment as previous, would you want to mark this as x-ms-pageable?

kc786 replied:
pushed my changes

veronicagg reviewed (changes requested):
Left a couple of comments inline. Rest looks good from my side. 
Adding @ravbhatnagar for ARM sign-off as PR adds 2 operations to the spec.

kc786 committed:
1. Added enum for grain field
2. added  x-ms-pageable to reservation summaries and reservation details operations and set set nextLinkName to null

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -155,7 +155,8 @@
             "required": true,
-            "type": "string"
+            "type": "string",
+            "enum": ["daily", "monthly"]
```
please take a look at adding https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum and most likely modelAsString=true

kc786 replied:
pushed the change

veronicagg reviewed (changes requested):
left one more comment, rest looks good from my side. 
Waiting on @ravbhatnagar for approval.

kc786 committed:
added modelAsString=true for enum field

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -156,7 +156,8 @@
             "type": "string",
-            "enum": ["daily", "monthly"]
+            "enum": ["daily", "monthly"],
+            "modelAsString":true
```
please take a look at https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum
there's a specific format to follow to add the x-ms-enum extension.

kc786 replied:
Done

veronicagg reviewed (changes requested):
left comment inline = please review format for https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum

kc786 committed:
corrected format to add x-ms-enum extension.

kc786 reviewed (commented):
corrected the format to add  x-ms-enum extension

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -157,7 +157,21 @@
+              "name": "datagrain",
+              "modelAsString":true,
+              "values":[
+                        { 
```
minor: formatting/indenting seems off, if you could use 2 spaces for formatting it'd be great.

kc786 replied:
Done

veronicagg commented on specification/consumption/resource-manager/Microsoft.Consumption/2017-11-30/consumption.json:
```diff
@@ -157,7 +157,21 @@
+                          "description":"Daily grain of data",
+                          "name":"DailyGrain"                          
+                        },
+                        { "value":"monthly",
```
minor: if you could be consistent with formatting here, and adding a new line after the "{", it'd be great.

kc786 replied:
Done

ravbhatnagar commented:
Looks good. Just camelcase the resource type names - reservationDetails and reservationSummaries

kc786 committed:
1. camel case operation names
2. Applied formating to document

kc786 committed:
Changed external docs  link

veronicagg reviewed (approved):
Thanks @kc786 and @ravbhatnagar . Looks good to me!

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby.svg?branch=RestAPI-PR2106)](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby) https://github.com/Azure/azure-sdk-for-ruby/pull/1176

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR2106)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/1689
